### PR TITLE
[WIP][DoNotMerge] 🌱 Bump controller-runtime to a17fd5870b906b50662fed6658265f5f8cf0863e / Generic fork

### PR DESCRIPTION
**Pull request by chrischdi**
Just testing this PR:

- https://github.com/kubernetes-sigs/controller-runtime/pull/2783

### CODE REVIEW

**Comment by chrischdi** [comment]
cc @sbueringer FYI

**Comment by sbueringer** [comment]
@chrischdi Thank you very much

Can you rebase when you have some time?

**Comment by sbueringer** [comment]
(also to reduce the delta now that we have Go 1.22 merged)

**Comment by sbueringer** [comment]
@chrischdi Thank you very very much for trying this out!

cc @alvaroaleman

File: controllers/external/tracker.go [hunk 1]
```diff
@@ -42,7 +41,7 @@ type ObjectTracker struct {
 }
 
 // Watch uses the controller to issue a Watch only if the object hasn't been seen before.
-func (o *ObjectTracker) Watch(log logr.Logger, obj runtime.Object, handler handler.EventHandler, p ...predicate.Predicate) error {
+func (o *ObjectTracker) Watch(log logr.Logger, obj client.Object, handler handler.TypedEventHandler[client.Object], p ...predicate.TypedPredicate[client.Object]) error {
```
Comment (sbueringer) [inline controllers/external/tracker.go:44]: Q: Could this diff be even smaller by keep using Predicate etc.? (type Predicate = TypedPredicate[client.Object])

**Comment by chrischdi** [comment]
/close

Will continue on #10383

**Pull request closed without merging**